dbms/src/DataStreams/ParallelAggregatingBlockInputStream.cpp

```diff
@@ -198,8 +198,8 @@ void ParallelAggregatingBlockInputStream::Handler::onException(std::exception_pt
 
     /// can not cancel parent inputStream or the exception might be lost
     if (!parent.executed)
-        /// kill the processor so ExchangeReceiver will be closed
-        parent.processor.cancel(true);
+        /// use cancel instead of kill to avoid too many useless error message
+        parent.processor.cancel(false);
```

Comment:
Cancel(true) is introduced by fixing  ExchangeReceiver not abort issue, now, we change back because we abort receivers in MPPTask top level?

Reply:
Just curious: why does TiRemoteBlockInputStream only cancel when kill is true?
https://github.com/pingcap/tiflash/blob/69cbfdf8a6bfb1d98ac76dea6e70d87ab3a1ed84/dbms/src/DataStreams/TiRemoteBlockInputStream.h#L200

Reply:
> 

Because sometimes `cancel(false)` is used to stop the query silently, for example, if multiple data streams is reading from `SharedQueryBlockInputStream`, if any of the data stream read an empty Block, it will call `cancel(false)` to stop other data streams that is reading from the same `SharedQueryBlockInputStream`, in this case we don't expect to get an error because of `cancel(false)`. But if `RemoteReader::cancel()`, it always throws an exception. We can skip this check If we add `is_kill` argument to `RemoteReader::cancel()` to support silent stop.